src/db/util/sequelize-model.ts

```diff
@@ -182,12 +183,27 @@ export const defineSequelizeModel =
       });
     };
 
+    const destroy: DestroyFn<Fields> = async (args) => {
+      if (opts.softDeletionEnabled) {
+        return (
+          await update({
+            ...args,
+            values: {
+              deletedAt: masterConn.fn.now(3),
+            } as unknown as Partial<UserDataOf<Fields>>,
+          })
+        ).length;
+      }
+      return await model.destroy(args);
+    };
+
```

Comment:
I think this could be a good solution for this.
Using 'function overload' while defining a new destroy function type could help on having this approach, validate the 'values' and the needs from the update function.
```suggestion
    const destroy: DestroySequelizeFn<Fields> = async (args) => {
      if (
        opts.softDeletionEnabled &&
        'values' in args
      ) {
        return (
          await model.update({
            ...args,
            values: {
              ...args.values,
              deletedAt: masterConn.fn.now(3),
              updatedAt: masterConn.fn.now(3),
            },
          })
        ).length;
      }
      return await model.destroy(args);
    };
```

Reply:
Under what circumstances would the destroy function ever be invoked with `values` within the args parameter? :P
I got an idea while looking at your suggestion. It could be a little better. Thanks for the hint and efforts.

Reply:
It's because you where using the 'update' function from the Sequelize wrapper, so I keep it. 
Glad it help